### PR TITLE
gg-jj: init at 0.23.0

### LINKED ISSUE #299856: Package request: gg
**Project description**

`gg` is a GUI for jj users. It is a Tauri app.

I believe that @Khionu is working on some stuff to assist packaging Tauri apps in nixpkgs. I wanted to open a 'tracking' issue, maybe to coordinate efforts and keep an eye on things, rather than constantly bugging them in the JJ Discord.

(jj is awesome, if you're reading this, you might find it neat to checkout!)

**Metadata**

* homepage URL: https://github.com/gulbanana/gg
* source URL: https://github.com/gulbanana/gg
* license: apl2
* platforms: unix, linux, darwin, ... ??

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pluiedev**
Closes #299856

It actually works this time.
image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gg/gg-jj/package.nix [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  # FIXME: Switch back to cargoHash when https://github.com/NixOS/nixpkgs/issues/356811 is fixed
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) pname version src;
+    sourceRoot = "${finalAttrs.src.name}/${finalAttrs.cargoRoot}";
+    hash = "sha256-Lr/0GkWHvfDy/leRLxisuTzGPZYFo2beHq9UCl6XlDo=";
+
+    nativeBuildInputs = [
+      yq
+      moreutils
```
Comment (getchoo) [inline pkgs/by-name/gg/gg-jj/package.nix:47]:
```suggestion
```

File: pkgs/by-name/gg/gg-jj/package.nix [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  nativeInstallCheckInputs = [ versionCheckHook ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "GUI for the version control system Jujutsu";
+    homepage = "https://github.com/gulbanana/gg";
+    changelog = "https://github.com/gulbanana/gg/blob/${finalAttrs.src.rev}/CHANGELOG.md";
+    license = with lib.licenses; [ asl20 ];
+    platforms = with lib.platforms; linux ++ darwin ++ windows;
```
Comment (getchoo) [inline pkgs/by-name/gg/gg-jj/package.nix:91]:
```suggestion
    platforms = with lib.platforms; linux ++ darwin;
```

`cargo-tauri.hook` only supports macOS and Linux
We could also probably inherit the platforms from there anyways

File: pkgs/by-name/gg/gg-jj/package.nix [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    cargo
+    rustc
+    nodejs
+    npmHooks.npmConfigHook
+    pkg-config
+    wrapGAppsHook3
+  ];
+
+  buildInputs =
+    [ openssl ]
```
Comment (getchoo) [inline pkgs/by-name/gg/gg-jj/package.nix:74]:
```suggestion
    [ openssl ]
      ++ lib.optional stdenv.hostPlatform.isDarwin apple-sdk_11
```
Fixes the [build error](https://logs.ofborg.org/?key=nixos/nixpkgs.360354&attempt_id=4702ae85-716f-45c8-b1d3-2c1188593fed) on x86_64-darwin

**Review by getchoo — DISMISSED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build gg-jj

**Comment by getchoo** [comment]
> Version check fails on darwin (because it generates darwin bundle?) Please add `optionals`

It would probably be better to link the executable in the bundle to $out/bin, similar to [modrinth-pp](https://github.com/NixOS/nixpkgs/blob/ac35b104800bff9028425fec3b6e8a41de2bbfff/pkgs/by-name/mo/modrinth-app-unwrapped/package.nix#L65). This would also allow for `lib.getExe`/meta.mainProgram/`nix run` to work correctly on Mac

**Comment by pluiedev** [comment]
> It would probably be better to link the executable in the bundle to $out/bin
I don't really have a Mac machine to check where the binary is in the bundle 😅

**Comment by Aleksanaa** [comment]
I'll update your PR later

**Comment by Aleksanaa** [comment]
@ofborg build gg-jj

**Pull request merged**